### PR TITLE
Speed up ci PR validation

### PR DESCRIPTION
**Pull request by johanandren**
References #523

Starting point ~35 minutes

### CODE REVIEW

**Comment by johanandren** [comment]
~26 minutes by not depending on the doc generation/validation

**Comment by johanandren** [comment]
~14 minutes (== check + slowest sample test run in sample-replicated-entity-examples)

**Comment by johanandren** [comment]
Down to ~12 minutes which essentially is 2 minutes code check and 10 minutes doc build.

Some curious things though: the set SDK version command takes ~1 minute in all jobs even though it only starts sbt to print a version. One thing we could do is to write it to a file in the workspace from the publish task, and read that rather than start sbt in all the sample builds. Probably not worth it as long as the doc build takes so long time.

**Comment by johanandren** [comment]
BTW this is a good view looking at how I parallelized the build: https://app.circleci.com/pipelines/github/lightbend/akkaserverless-java-sdk/2000/workflows/95044ee6-2025-4a68-b010-43ff2db1ab91

**Review by octonato — APPROVED** [review]
Nice!!

**Comment by pvlugter** [comment]
Nice improvement by parallelising 👍🏼 

> slowest sample test run in sample-replicated-entity-examples

It has a separate integration test (with proxy docker container) for each replicated data type. But probably not worthwhile breaking it up if the docs are longer.

> 10 minutes doc build

The docs build does the external link validation now. That could be a separate job to parallelise this.

> Some curious things though: the set SDK version command takes ~1 minute in all jobs even though it only starts sbt to print a version. One thing we could do is to write it to a file in the workspace from the publish task, and read that rather than start sbt in all the sample builds.

Could also run the appropriate `git describe` command directly — but would need to be aligned with whatever sbt-dynver is doing. There's a scripted version in the JavaScript SDK:

https://github.com/lightbend/akkaserverless-javascript-sdk/blob/458b48f745c5f0587f7f30ab057eb1644b8b1376/sdk/bin/version.sh#L22-L39

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,14 +78,30 @@ commands:
     steps:
       - run:
           name: publish m2 and local
-          command: sbt publishM2 publishLocal
+          command: sbt publishM2 publishLocal -Ddisable.apidocs=true
```
Comment (patriknw) [inline .circleci/config.yml:81]: That's a good one. Doc generation is incredible slow. That is probably what takes most time of the doc Make.

**Comment by johanandren** [comment]
Extracting the link validation didn't make a difference but maybe worth it anyway to get a smaller/quicker job to re-run when some link fails temporarily.

**Review by patriknw — APPROVED** [review]
LGTM, that should make a big difference

**Comment by pvlugter** [comment]
> Extracting the link validation didn't make a difference but maybe worth it anyway to get a smaller/quicker job to re-run when some link fails temporarily.

Ah, that doesn't as long as I thought it might. Makes sense to have separate as it could be flaky.

**Review by pvlugter — APPROVED** [review]

**Pull request merged**